sdk/src/Core/Amazon.Util/AWSSDKUtils.cs

```diff
@@ -427,25 +427,32 @@ public static string CanonicalizeResourcePath(Uri endpoint, string resourcePath,
         /// <returns>Canonicalized resource path for the endpoint.</returns>
         public static string CanonicalizeResourcePathV2(Uri endpoint, string resourcePath, bool encode, IDictionary<string, string> pathResources)
         {
+            // Single encode the resourcepath from the endpoint i.e. treat it like a regular uri.
+            // Double encode the resource path from the request
+            string resourcePathFromRequest = resourcePath != null ? resourcePath : string.Empty;
```

Comment:
After this variable is set here can we only use `resourcePathFromRequest` instead of `resourcePath` anywhere else in the function? I like that we have 2 clearly named variables and still using the `resourcePath` leaves some confusion.